File: src/impl/ServerImpl.java
package impl;

import inter.ServerInterface;

import java.rmi.RemoteException;
import java.rmi.server.UnicastRemoteObject;
import java.util.ArrayList;
import java.util.Arrays;

public class ServerImpl extends UnicastRemoteObject implements ServerInterface {

    private ArrayList<String> message;

    public ServerImpl() throws RemoteException {
        super();
        message = new ArrayList<>(Arrays.asList("test1","test2"));
    }

    @Override
    public ArrayList<String> getMessages() throws RemoteException {
        return message;
    }

    @Override
    public void send(String str) {
        message.add(str);
    }
}

File: src/impl/ServerImpl.kt
package impl

import inter.ServerInterface
import metier.Message

import java.rmi.RemoteException
import java.rmi.server.UnicastRemoteObject
import java.time.LocalDate
import java.util.ArrayList
import java.util.Arrays

class ServerImpl @Throws(RemoteException::class)
constructor() : UnicastRemoteObject(), ServerInterface {

    override
    val messages = ArrayList(Arrays.asList(Message("toto", LocalDate.now(), "tes1"),
                Message("toto", LocalDate.now(), "tes2")))

    @Throws(RemoteException::class)
    override
    fun send(pseudo: String, message: String) {
        this.messages.add(Message(pseudo, LocalDate.now(), message))
    }
}
